### PR TITLE
🏗️ upgrade all the things 🤖

### PR DESCRIPTION
**Pull request by bollwyvl**
attempt for #40... more to come

### CODE REVIEW

**Comment by bollwyvl** [comment]
first green checkmark (linux):
https://dev.azure.com/robots-from-jupyter/robots-from-jupyter/_build/results?buildId=156

**Comment by bollwyvl** [comment]
I need to get my local windows build environment working again, which might take some time...

**Comment by bollwyvl** [comment]
Lot of CI flailing here, glad it's free, and sorry if this has been a painful thing to watch.

Looks like Linux and OSX are squared away.

On windows, it's now finally _gotten_ to the acceptance tests, but no idea what we'll see... i'll probably split up some more of the package building for windows so that we're only building `robotlab` in the critical path.

**Comment by bollwyvl** [comment]
We could save a few more minutes up front by only _building_ conda packages in the critical path, and then _testing_ them as parallel jobs while the other stuff is going on, or deferring them until we have An Answer from the acceptance tests. It basically costs 1 minute (or 3, on windows) to spin up another job, and getting it under half an hour would be really nice.

**Comment by bollwyvl** [comment]
Have gotten all the way to this error on windows in acceptance tests:

selenium-screenshot-1

turns out:
```
(%PREFIX%) %SRC_DIR%>copy app_dir %PREFIX%\\share\\jupyter\\robotlab 
app_dir\imports.css
        1 file(s) copied.
```

doh. now using `xcopy`, but might go to a python script if i keep screwing these up...

**Comment by bollwyvl** [comment]
WHUT

https://dev.azure.com/robots-from-jupyter/robots-from-jupyter/_build/results?buildId=175

**Comment by bollwyvl** [comment]
> before: 36:41.300
> after: 35:16.421

the windows build of two packages now takes longer than the four noarch packages. sigh. The linux and osx builds start immediately.
 
Splitting the windows builds even further seems a bit silly, but could buy another two minutes, probably, as the bottom bound is 3 minutes to install the build env and another 3 to build the hardest package.

Running the full conda-build tests against `robotlab` (especially on windows) is unfortunately too useful to postpone, as it should now be quite good at capturing any issues that previously might not occur until the constructor solve, or worse yet, buried in a log file during acceptance tests when the installer is actually running. So 35 minutes it is!

I guess the next step is to take some of these entirely new-fangled beasts and actually download and _use_ them a bit, though we could go ahead and merge (squash, of course, to protect the guilty). Onward!

**Comment by datakurre** [comment]
Thanks. I already had to build at https://github.com/robots-from-jupyter/robotlab/pull/41#issuecomment-531612728 on Windows, but I could not figure out how to start RobotLab. Trying again.

https://dev.azure.com/robots-from-jupyter/robots-from-jupyter/_build/results?buildId=179&view=logs&jobId=2d2b3007-3c5c-5840-9bb0-2b1ea49925f3

**Comment by datakurre** [comment]
Yep. Unfortunately it looks that things are not ok yet in Windows land at least. post_install.bat looks like Linux/macOS-version and is not run. None of the Scripts under robotlab-package runs and I am unable to create even Conda / Anaconda prompt. Only binary created directly under the installation directory, is `_conda.exe`.

**Comment by bollwyvl** [comment]
Thanks for the testing! I'll keep investigating, too!

**Comment by datakurre** [comment]
To put your hard work in use, I did submit a rerun of the RobotLab workshop for the next Robocon. (They have DL for proposals today.) And no, I don’t expect you to fly here every year 😅

**Comment by bollwyvl** [comment]
> They have DL for proposals today

Ha, flaked on that, had some ideas around VM/k8s robot stuff, but it's good! I'd happily help out remotely :P

**Comment by datakurre** [comment]
This happened on Linux
```
Welcome to RobotLab 2019.9.0

RobotLab will now be installed into this location:
/home/datakurre/robotlab

  - Press ENTER to confirm the location
  - Press CTRL-C to abort the installation
  - Or specify a different location below

[/home/datakurre/robotlab] >>> 
PREFIX=/home/datakurre/robotlab
Unpacking payload ...
Extracting : robotframework-seleniumlibrary-3.3.1-py_0.tar.bz2:  89%|########8 | 132/149 [00:33<00:12,  1.32it/Extracting : robotkernel-1.0-py_0.tar.bz2:  89%|########9 | 133/149 [00:33<00:12,  1.32it/s]                   Extracting : robotframework-jupyterlibrary-0.2.0-py_0.tar.bz2:  91%|#########1| 136/149 [00:33<00:09,  1.32it/sExtracting : robotframework-seleniumscreenshots-0.9.5-py_0.tar.bz2:  92%|#########1| 137/149 [00:33<00:09,  1.3Extracting : widgetsnbextension-3.5.1-py36_0.tar.bz2:  93%|#########2| 138/149 [00:33<00:08,  1.32it/s]        Collecting package metadata (current_repodata.json): failed                                            

UnavailableInvalidChannel: The channel is not accessible or is invalid.
  channel name: conda-bld
  channel url: file:///home/vsts/work/1/s/_artifacts/conda-bld
  error code: 404

You will need to adjust your conda configuration to proceed.
Use `conda config --show channels` to view your configuration's current state,
and use `conda config --show-sources` to view config file locations.

As of conda 4.3, a valid channel must contain a `noarch/repodata.json` and
associated `noarch/repodata.json.bz2` file, even if `noarch/repodata.json` is
empty. Use `conda index /home/vsts/work/1/s/_artifacts/conda-bld`, or create `noarch/repodata.json`
and associated `noarch/repodata.json.bz2`.
```
```
~/robotlab$ ls -1
conda-meta
conda.exe
pkgs
postconda.tar.bz2
```

**Comment by bollwyvl** [comment]
Indeed, i'll yet to finish my windows vm setup, too, so can try out some of this stuff locally.

But the above error, i know how to fix (one-liner in `constructor.yaml`)... and test it: we run the tests from inside the installer on a clean machine. How often do you get to do that? :spoon: I guess the full meta would be to port them to notebooks... one bootstrap notebook started with `nbrobot` which in turn ran all the other notebooks. Yeesh.

**Comment by datakurre** [comment]
Quoting Nicholas Bollweg (2019-09-18 02:05:16)
> we run the tests from inside the installer on a clean machine.
> How often do you get to do that?

o.0? Like Nix that does build and test every package in an isolated
sandbox by default? (But sure does not run Windows :/)

**Comment by bollwyvl** [comment]
> Like Nix that does build and test every package in an isolated sandbox by default

yeah, conda, too, to the best of its abilities, given the other constraints (windows).

> But sure does not run Windows 

right, that's what we ran afoul of: everything is special on windows. there was a dangling pointer to the absolute path of the built artifact... it could find it in CI we're just pretending you _could_ get that from `conda-forge`... an indeed, it emits warnings during the build.

>> we run the tests from inside the installer on a clean machine.
>> How often do you get to do that?

What i mean is, it is infrequent to actually ship something that also includes all of its test dependencies: usually you don't want those. in this case, we now ship firefox, geckodriver, conda, jupyter++, robotframework++, and python on three platforms... which is exactly what we need to test testing firefox, geckodriver, conda, jupyter++, robotframework++, and python on three platforms: prior to the `${IN_ROBOTLAB}` stuff, we weren't actually testing whether our bundled ff/gd/etc. were actually working. we now don't even install the test stuff in the outermost test environment.

Obviously still some issues... :grimacing:

File: tests/resources/Install.robot [hunk 1]
```diff
@@ -66,7 +66,7 @@ Run the RobotLab Windows installer
     ${installer} =    Set Variable    ${INSTALLER DIR}${/}RobotLab-${INSTALLER VERSION}-Windows-x86_64.exe
     ${args} =    Set Variable    /InstallationType=JustMe /AddToPath=0 /RegisterPython=0 /S /D=${ROBOTLAB DIR}
     ${result} =    Run Keyword If    not ${IN_ROBOTLAB}
-    ....    Run Process    ${installer} ${args}    stdout=${INSTALL LOG}    stderr=STDOUT    shell=True    env:HOME=${FAKE HOME}
```
Comment (bollwyvl) [inline tests/resources/Install.robot]: :man_facepalming:

Comment (bollwyvl) [reply tests/resources/Install.robot]: to make matters worse, i've done that before: should add a linter rule...

**Comment by bollwyvl** [comment]
Most recent pass is looking 🐸 but maybe wait until I test out this evening
on Linux/windows before sinking any time into it. Bottom line from this
adventure is I basically don't understand windows home directories 😭.

Also definitely points to doing some more work on rfjl to account for some
of these browser slowness things.

Looking forward to releasing this before everything puts out another
version!

**Comment by datakurre** [comment]
For a motivation, Google found me a thesis that promoted RobotLab:

>The biggest criticism against Robot Framework is that it usually requires a complicated installing process. This problem has been tackled with RobotLab package that is very easy to install

> After trying up few different setups ofRobot Framework combined with variety of GUI (graphical user interface) tools it turned out that there was a much easier way to do it. By installing Robotlab

https://www.theseus.fi/bitstream/handle/10024/169008/Tiia-Rautavesi-thesis.pdf?sequence=2&isAllowed=y

**Comment by bollwyvl** [comment]
> The only downside tothis is the inconvenient package size (400-500MB). 

Guess we should offer a RobotLabLite (not on this PR :blush: )? Here's a `du` of what we're currently building (on linux):
<details>
<pre>
8	blas-1.0-mkl.tar.bz2
8	validate_email-1.3-py_3.tar.bz2
12	entrypoints-0.3-py36_0.tar.bz2
12	jsonpointer-2.0-py_0.tar.bz2
12	opencv-3.4.2-py36h6fd60c2_1.tar.bz2
12	robotframework-seleniumscreenshots-0.9.5-py_0.tar.bz2
16	libuuid-1.0.3-h1bed415_2.tar.bz2
16	pandocfilters-1.4.2-py36_1.tar.bz2
16	pickleshare-0.7.5-py36_0.tar.bz2
16	send2trash-1.5.0-py36_0.tar.bz2
20	backcall-0.1.0-py36_0.tar.bz2
20	decorator-4.4.0-py36_1.tar.bz2
20	genson-1.1.0-py_0.tar.bz2
20	robotframework-jupyterlibrary-0.2.0-py_0.tar.bz2
20	webencodings-0.5.1-py36_1.tar.bz2
24	defusedxml-0.6.0-py_0.tar.bz2
24	ptyprocess-0.6.0-py36_0.tar.bz2
24	python-libarchive-c-2.8-py36_13.tar.bz2
24	six-1.12.0-py36_0.tar.bz2
24	strict-rfc3339-0.7-py_1.tar.bz2
24	terminado-0.8.2-py36_0.tar.bz2
28	json5-0.8.5-py_0.tar.bz2
28	jsonpath-ng-1.4.3-py_1.tar.bz2
28	jupyterlab_server-1.0.0-py_1.tar.bz2
28	restinstance-1.0.2-py_0.tar.bz2
28	rfc3987-1.3.8-py_0.tar.bz2
28	tzlocal-2.0.0-py36_0.tar.bz2
28	wcwidth-0.1.7-py36_0.tar.bz2
32	markupsafe-1.1.1-py36h7b6447c_0.tar.bz2
32	pysocks-1.7.0-py36_0.tar.bz2
40	ipython_genutils-0.2.0-py36_0.tar.bz2
40	wheel-0.33.4-py36_0.tar.bz2
44	libffi-3.2.1-h4deb6c0_3.tar.bz2
44	prometheus_client-0.7.1-py_0.tar.bz2
48	flex-swagger-6.14.0-py_0.tar.bz2
48	jupyter_core-4.5.0-py_0.tar.bz2
48	tqdm-4.32.1-py_0.tar.bz2
52	numpy-1.16.5-py36h7e9f1db_0.tar.bz2
52	olefile-0.46-py36_0.tar.bz2
56	mistune-0.8.4-py36h7b6447c_0.tar.bz2
56	robotframework-seleniumlibrary-3.3.1-py_0.tar.bz2
56	robotkernel-1.0-py_0.tar.bz2
60	attrs-19.1.0-py36_1.tar.bz2
68	parso-0.5.1-py_0.tar.bz2
72	jupyter_client-5.3.1-py_0.tar.bz2
80	ply-3.11-py36_0.tar.bz2
84	pexpect-4.7.0-py36_0.tar.bz2
84	pyopenssl-19.0.0-py36_0.tar.bz2
88	yaml-0.1.7-h96e3832_1.tar.bz2
92	jsonschema-3.0.2-py36_0.tar.bz2
92	pyrsistent-0.14.11-py36h7b6447c_0.tar.bz2
92	requests-2.22.0-py36_0.tar.bz2
92	testpath-0.4.2-py36_0.tar.bz2
104	graphite2-1.3.13-h23475e2_0.tar.bz2
104	pycosat-0.6.3-py36h14c3975_0.tar.bz2
108	bzip2-1.0.8-h7b6447c_0.tar.bz2
108	ipywidgets-7.5.1-py_0.tar.bz2
116	pyshortcuts-1.4-py36_0.tar.bz2
120	click-7.0-py36_0.tar.bz2
120	zlib-1.2.11-h7b6447c_3.tar.bz2
136	ca-certificates-2019.5.15-1.tar.bz2
136	idna-2.8-py36_0.tar.bz2
136	traitlets-4.3.2-py36_0.tar.bz2
144	nbformat-4.4.0-py36_0.tar.bz2
156	asn1crypto-0.24.0-py36_0.tar.bz2
156	certifi-2019.6.16-py36_1.tar.bz2
156	urllib3-1.24.2-py36_0.tar.bz2
160	lz4-c-1.8.1.2-h14c3975_0.tar.bz2
168	ipykernel-5.1.2-py36h39e3cac_0.tar.bz2
176	mkl_fft-1.0.14-py36ha843d7b_0.tar.bz2
176	pycparser-2.19-py36_0.tar.bz2
188	libedit-3.1.20181209-hc058e9b_0.tar.bz2
192	jinja2-2.10.1-py36_0.tar.bz2
192	pyyaml-5.1.2-py36h7b6447c_0.tar.bz2
200	chardet-3.0.4-py36_1003.tar.bz2
208	mkl-service-2.3.0-py36he904b0f_0.tar.bz2
224	cffi-1.12.3-py36h2e261b9_0.tar.bz2
228	bleach-3.1.0-py36_0.tar.bz2
248	jpeg-9b-habf39ab_1.tar.bz2
248	pytz-2019.2-py_0.tar.bz2
248	ruamel_yaml-0.15.46-py36h14c3975_0.tar.bz2
252	freeglut-3.0.0-hf484d3e_5.tar.bz2
264	conda-package-handling-1.3.11-py36_0.tar.bz2
264	pcre-8.43-he6710b0_0.tar.bz2
284	python-dateutil-2.8.0-py36_0.tar.bz2
292	fontconfig-2.13.0-h9420a91_0.tar.bz2
304	libsodium-1.0.16-h1bed415_0.tar.bz2
316	lzo-2.10-h1bfc0ba_1.tar.bz2
368	libpng-1.6.37-hbc83047_0.tar.bz2
368	xz-5.2.4-h14c3975_4.tar.bz2
380	libglu-9.0.0-hf484d3e_1.tar.bz2
384	nbconvert-5.5.0-py_0.tar.bz2
392	readline-7.0-h7b6447c_5.tar.bz2
408	mkl_random-1.0.2-py36hd81dba3_0.tar.bz2
496	prompt_toolkit-2.0.9-py36_0.tar.bz2
504	libxcb-1.13-h1bed415_1.tar.bz2
520	pyzmq-18.1.0-py36he6710b0_0.tar.bz2
604	libtiff-4.0.10-h2733197_2.tar.bz2
620	cryptography-2.7-py36h1ba5d50_0.tar.bz2
620	pixman-0.38.0-h7b6447c_0.tar.bz2
632	libopus-1.3-h7b6447c_0.tar.bz2
644	pillow-6.1.0-py36h34e0f95_0.tar.bz2
644	tornado-6.0.3-py36h7b6447c_0.tar.bz2
660	setuptools-41.0.1-py36_0.tar.bz2
664	pygments-2.4.2-py_0.tar.bz2
668	zeromq-4.3.1-he6710b0_3.tar.bz2
704	future-0.17.1-py36_0.tar.bz2
724	jedi-0.15.1-py36_0.tar.bz2
744	docutils-0.15.2-py36_0.tar.bz2
744	gmp-6.1.2-hb3b607b_0.tar.bz2
824	freetype-2.9.1-h8a8886c_1.tar.bz2
864	harfbuzz-1.8.8-hffaf4a1_0.tar.bz2
864	selenium-3.141.0-py36h7b6447c_0.tar.bz2
872	conda-package-handling-1.5.0-py36h7b6447c_0.tar.bz2
880	intel-openmp-2019.4-243.tar.bz2
888	zstd-1.3.7-h0b5b093_0.tar.bz2
956	robotframework-3.1.2-py_0.tar.bz2
960	ncurses-6.1-he6710b0_1.tar.bz2
1028	lunr-0.5.5-py36_0.tar.bz2
1140	ipython-7.8.0-py36h39e3cac_0.tar.bz2
1160	jasper-2.0.14-h07fcdf6_1.tar.bz2
1244	py-opencv-3.4.2-py36hb342d67_1.tar.bz2
1308	libgfortran-ng-7.3.0-hdf63c60_0.tar.bz2
1332	cairo-1.14.12-h8948797_3.tar.bz2
1532	libarchive-3.3.3-h5d8350f_5.tar.bz2
1856	widgetsnbextension-3.5.1-py36_0.tar.bz2
1948	sqlite-3.29.0-h7b6447c_0.tar.bz2
1960	pip-19.2.2-py36_0.tar.bz2
2024	libxml2-2.9.9-hea5a465_1.tar.bz2
2448	libvpx-1.7.0-h439df22_0.tar.bz2
2576	geckodriver-0.24.0-hf484d3e_1.tar.bz2
3088	conda-4.7.12-py36_0.tar.bz2
3220	tk-8.6.8-hbc83047_0.tar.bz2
4148	libstdcxx-ng-9.1.0-hdf63c60_0.tar.bz2
4488	numpy-base-1.16.5-py36hde5b4d6_0.tar.bz2
5108	glib-2.56.2-hd408876_0.tar.bz2
5148	openssl-1.1.1-h7b6447c_0.tar.bz2
5328	hdf5-1.10.2-hba1933b_1.tar.bz2
6192	notebook-6.0.1-py36_0.tar.bz2
8300	libgcc-ng-9.1.0-hdf63c60_0.tar.bz2
11944	jupyterlab-1.1.3-pyhf63ae98_0.tar.bz2
13540	robotlab-2019.9.0-py36_0.tar.bz2
16992	nodejs-11.14.0-he1b5a44_1.tar.bz2
21272	pandoc-2.2.3.2-0.tar.bz2
23068	icu-58.2-h211956c_0.tar.bz2
35268	python-3.6.9-h265db76_0.tar.bz2
41332	libopencv-3.4.2-hb342d67_1.tar.bz2
62468	firefox-68.0.2-0.tar.bz2
75480	ffmpeg-4.0-hcdf2ecd_0.tar.bz2
208984	mkl-2019.4-243.tar.bz2

</pre>
</details>

Adding `firefox` certainly didn't help matters! Cutting `firefox`, `pandoc`, `opencv`/`pillow` (and therefore `mkl`, `numpy` and `ffmpeg`) would probably end up being pretty light. 

Since we ship our own `jupyterlab` assets, we could probably repackage without its assets.

**Comment by bollwyvl** [comment]
Windows testing looks great. Installer worked, shortcut created, launched in firefox. The widgets work great.

Started doing the tutorials: launching lab worked (into the bundled firefox!).

Did get some errors
<details>
<pre>
 File "C:\Users\IEUser\RobotLab\lib\site-packages\robotkernel\kernel.py", line 269, in do_execute
    silent,
  File "C:\Users\IEUser\RobotLab\lib\site-packages\robotkernel\executors.py", line 191, in execute_robot
    reply = run_robot_suite(kernel, suite, listeners, silent, display_id, path)
  File "C:\Users\IEUser\RobotLab\lib\tempfile.py", line 809, in __exit__
    self.cleanup()
  File "C:\Users\IEUser\RobotLab\lib\tempfile.py", line 813, in cleanup
    _shutil.rmtree(self.name)
  File "C:\Users\IEUser\RobotLab\lib\shutil.py", line 500, in rmtree
    return _rmtree_unsafe(path, onerror)
  File "C:\Users\IEUser\RobotLab\lib\shutil.py", line 395, in _rmtree_unsafe
    onerror(os.unlink, fullname, sys.exc_info())
  File "C:\Users\IEUser\RobotLab\lib\shutil.py", line 393, in _rmtree_unsafe
    os.unlink(fullname)
PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\IEUser\\AppData\\Local\\Temp\\tmpi05hwbse\\geckodriver.log'
</pre>
</details>

Whitelibrary didn't seem to be doing the thing.

Getting this a lot:
<details>
<pre>
[ ERROR ] Calling method 'start_suite' of listener 'RobotVariablesListener' failed: Dictionary item '' does not contain '=' separator.
</pre>
</details>

So there are still some rough edges, but all in all, really beautiful stuff, man. Will take a spin on Linux!

**Comment by bollwyvl** [comment]
On linux:
Yeah, surprise, no working shortcut icon. Boo. Probably have to land it upstream or something.

Everything else works, still seeing the error...
```
[ ERROR ] Calling method 'start_suite' of listener 'RobotVariablesListener' failed: Dictionary item '' does not contain '=' separator.
```
...rather a lot.

The Lab 1.0 DOM has changed enough that a number of the tutorials are broken, so those probably need a tune-up. Google also changed some stuff... lots more autocomplete, and their completion choices cover up the search button... I do always love giving that demo and localizing it on the fly for english. Also on the DOM piece, I did encounter some issues with the `%%python` syntax highlighting, which seems like it's probably my fault :nerd_face: 

Really happy with what I'm seeing! We still need to do something about the discoverability of the tutorials and examples (#3)... but I don't think we want to add any more bells and whistles before getting at least a Beta out... it's your call @datakurre whether we hold to update the tutorials/examples!

**Comment by bollwyvl** [comment]
I'm happy to roll back the lab version to `jupyterlab==1.1.3`, but am trying some things locally to overload the icon selectors... in-line `Import Resource`? variable file?

that's all I anticipate changing, looking at the changelog. However, getting a `robotframework-jupyterlibrary` release out isn't going to happen tonight :P

**Comment by datakurre** [comment]
Quoting Nicholas Bollweg (2019-09-19 03:22:18)
> Everything else works, still seeing the error...
> [ ERROR ] Calling method 'start_suite' of listener 'RobotVariablesListener' fail
> ed: Dictionary item '' does not contain '=' separator.

That's a robotkernel bug that I thought I already fixed. I try to look
into that again tomorrow.

That listeners tries to keep robot variable states between executions,
but robot BuiltIn library cannot fully import the format it exports.

> The Lab 1.0 DOM has changed enough that a number of the tutorials are
> broken, so those probably need a tune-up. Google also changed some
> stuff... lots more autocomplete, and their completion choices cover up
> the search button... I do always love giving that demo and localizing
> it on the fly for english.

That was expected. I should go them through before the final release.

**Comment by datakurre** [comment]
Quoting Nicholas Bollweg (2019-09-18 23:20:45)
> The only downside tothis is the inconvenient package size
> (400-500MB).

That's mostly psychological, I belive. Usually, if you are not downloading a big installer, the installer is downloading them for you.

A smaller installer alternative might be nice, but only if we can get
this fully automated up to GitHub releases :)

**Comment by bollwyvl** [comment]
Oooh, back in business with a hot patch during build. Gross but beautiful. Thanks for bearing with me on this.

> I should go them through before the final release.

I'll agree with you, but because #3, it's hard to even find them... as long as the suggested way to get them is from the site, which is great, it probably doesn't even have to hold up merging this. I'll do another round of machine testing with the 1.1.4 versions, but like I said, everything is in mostly-working order. Given that we can't even really iterate on `master` makes it tough to untangle what should be done next.

> That's mostly psychological, I belive. Usually, if you are not downloading a big installer, the installer is downloading them for you.

Oh, for sure! And in the case where connectivity is poor, it's _far_ easier to hand around a big slug of data. We got lucky at the workshop, being in a place of business, but a lot of venues are Just Awful. But it is still worth exploring (once we get all the other stuff squared).

> A smaller installer alternative might be nice, but only if we can get this fully automated up to GitHub releases :)

Yep. Azure has their own long-term thing, but I don't like sending people there. It's kind of a mess. Getting the github connection to work on tags is definitely high priority, and I'll tackle that shortly... but probably not on this PR.

> That listeners tries to keep robot variable states between executions, but robot BuiltIn library cannot fully import the format it exports.

You are a magician. We should see about getting some of the things we've been doing (especially the in-memory readers, etc.) into core.

**Comment by datakurre** [comment]
I already tried Linux-version and it looked great. I should have time in the evening so I try to do my part (robotkernel 1.0.1 and tutorials) so that we can have a release once you feel like it.

**Comment by datakurre** [comment]
Oh, I agree on you that fixing tutorials should not block this pull. Even robotkernel 1.0.1 should not be a blocker on merging this (even it is blocker for a release tag). I'd be pleased to try if I can upgrade it on my own by just doing a new pull for robotlab.

**Comment by bollwyvl** [comment]
Ah, very good. I'll do that windows test against the most recent build and
merge if it looks good.

**Comment by bollwyvl** [comment]
Yeah, looks good on windows, merging. Will monitor build, but probably won't be able to react to much...

**Pull request merged**